google-cloud-spanner/src/main/java/com/google/cloud/spanner/GrpcStreamIterator.java

```diff
@@ -66,6 +68,10 @@ protected final SpannerRpc.ResultStreamConsumer consumer() {
     return consumer;
   }
 
+  public void registerListener(AsyncResultSet.StreamMessageListener streamMessageListener) {
+    this.streamMessageListener = streamMessageListener;
```

Comment:
nit:
```suggestion
    Preconditions.checkState(this.streamMessageListener == null);
    this.streamMessageListener = Preconditions.checkNotNull(streamMessageListener);
```

Reply:
this.streamMessageListener = Preconditions.checkNotNull(streamMessageListener); this alone is enough right?

Reply:
done